Source/CarthageKit/CompatibilityInfo.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+/// Identifies a dependency, its pinned version, and versions of this dependency with which it may or may not be compatible
+public struct CompatibilityInfo: Equatable {
+	/// The dependecy
+	public let dependency: Dependency
+
+	/// The pinned version of the dependency
+	public let pinnedVersion: PinnedVersion
+	
+	/// Versions of this dependency with which it may or may not be compatible
+	public let dependencyVersions: [(Dependency, VersionSpecifier)]
```

Comment:
I think this property could have a better name. It seems like this is actually the requirements for this dependency. And it probably makes more sense as a dictionary.

```swift
/// The projects that list this dependency as a requirement and the version specifiers they use.
public let requirements: [Dependency: VersionSpecifier]
```